modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.search.experiences.internal.upgrade.v3_1_2;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONException;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Felipe Lorenz
+ */
+public class SXPBlueprintUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeSXPElement();
+
+		_upgradeSXPBlueprint();
+	}
+
+	private JSONArray _createJSONArray(String jsonArrayString) {
+		try {
+			return JSONFactoryUtil.createJSONArray(jsonArrayString);
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	private JSONObject _createJSONObject(String jsonObjectString) {
+		try {
+			return JSONFactoryUtil.createJSONObject(jsonObjectString);
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	private String _replaceHelpTextInElementDefinition(
+		String elementDefinition) {
+
+		if (Validator.isBlank(elementDefinition)) {
+			return elementDefinition;
+		}
+
+		JSONObject sxpElementJSONObject = _createJSONObject(elementDefinition);
+
+		if (sxpElementJSONObject == null) {
+			return elementDefinition;
+		}
+
+		JSONObject uiConfigurationJSONObject =
+			sxpElementJSONObject.getJSONObject("uiConfiguration");
+
+		if (uiConfigurationJSONObject == null) {
+			return elementDefinition;
+		}
+
+		JSONArray fieldSetsJSONArray = uiConfigurationJSONObject.getJSONArray(
+			"fieldSets");
+
+		if (fieldSetsJSONArray == null) {
+			return elementDefinition;
+		}
+
+		for (int i = 0; i < fieldSetsJSONArray.length(); i++) {
+			JSONObject fieldSetJSONObject = fieldSetsJSONArray.getJSONObject(i);
+
+			JSONArray fieldsJSONArray = fieldSetJSONObject.getJSONArray(
+				"fields");
+
+			if (fieldsJSONArray == null) {
+				continue;
+			}
+
+			for (int j = 0; j < fieldsJSONArray.length(); j++) {
+				JSONObject fieldJSONObject = fieldsJSONArray.getJSONObject(j);
+
+				fieldJSONObject.put("helpText", "group-ids-help");
+			}
+		}
+
+		return sxpElementJSONObject.toString();
+	}
+
+	private String _replaceHelpTextInElementInstanceJSON(
```

Comment:
Let's do better - `_replaceHelpTextForLimitSearchToTheseSites`

Reply:
what about `_replaceHelpTextInElementDefinition`?

Reply:
To me it's important that the name communicates the code is only operating on LimitSearchToTheseSites Element Definitions.